### PR TITLE
Setup App

### PR DESCRIPTION
**Pull request by mohametalmeari**
- Create Rails App
- Connect App to database
- Install Linters

### CODE REVIEW

File: db/migrate/20230814185652_add_details_to_posts.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class AddDetailsToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```
Comment (Gambit142) [inline db/migrate/20230814185652_add_details_to_posts.rb:3]: - [ ] In the schema diagram, a `User` model has many `Post` models through the `Author` column. Therefore you need to implement it in such a way that the foreign key is the `author_id`. Kindly reflect this relationship in this migration file. 👍 😎 

#### Hint: Replace the line of code that links the `User` model to the `Post` model with the line of code below:
```ruby
add_reference :posts, :author, null: false, foreign_key: { to_table: :users }
```

File: db/migrate/20230814185657_add_details_to_likes.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class AddDetailsToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: true
```
Comment (Gambit142) [inline db/migrate/20230814185657_add_details_to_likes.rb:3]: - [ ] In the schema diagram, a `User` model has many `Like` models through the `Author` column. Therefore you need to implement it in such a way that the foreign key is the `author_id`. Kindly reflect this relationship in this migration file. 👍 😎 

#### Hint: Replace the line of code that links the `User` model to the `Like` model with the line of code below:
```ruby
add_reference :likes, :author, null: false, foreign_key: { to_table: :users }
```

File: db/migrate/20230814185740_add_details_to_comments.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class AddDetailsToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
```
Comment (Gambit142) [inline db/migrate/20230814185740_add_details_to_comments.rb:3]: - [ ] In the schema diagram, a `User` model has many `Comment` models through the `Author` column. Therefore you need to implement it in such a way that the foreign key is the `author_id`. Kindly reflect this relationship in this migration file. 👍 😎 

#### Hint: Replace the line of code that links the `User` model to the `Comment` model with the line of code below:
```ruby
add_reference :comments, :author, null: false, foreign_key: { to_table: :users }
```

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @mohametalmeari 👋, 

Good job so far!

## Highlights
- Setup linters correctly ✔️ 
- Created migration files for the `User` model ✔️ 
- Created migration files for the `Post` model ✔️ 
- Created migration files for the `Comment` model ✔️ 
- Created migration files for the `Like` model ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by mukhammadrizooff — APPROVED** [review]
Hi @mohametalmeari, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**